Sync/SyncStateMachine.swift

```diff
@@ -682,6 +718,24 @@ open class InitialWithLiveTokenAndInfo: BaseSyncStateWithInfo {
             log.debug("No cached meta/global found. Fetching fresh meta/global.")
         }
 
+        return deferMaybe(NeedsFreshMetaGlobal.fromState(self))
```

Comment:
Line 715 will run even if we subsequently refuse to enter `NeedsFreshMetaGlobal`. That is, we are doing some work when we know we're going to fail.

I haven't read closely enough to determine whether this could run twice as a result.

Could you make sure that (a) this is acceptable from an amount-of-work perspective, and (b) that we can pick up where we left off on the next sync, doing the exact same work if we run optimistic then non-optimistic as if we run just non-optimistic?

If not, you could/should add a check around line 714 that `NeedsFreshMetaGlobal` is in our set of acceptable states.